src/msg/cli.rs

```diff
@@ -469,6 +467,57 @@ fn msg_matches_reply(ctx: &Ctx, matches: &clap::ArgMatches) -> Result<bool> {
     Ok(true)
 }
 
+pub fn msg_matches_mailto(ctx: &Ctx, url: &Url) -> Result<()> {
+    debug!("mailto command matched");
+
+    let mut imap_conn = ImapConnector::new(&ctx.account)?;
+
+    let mut cc = Vec::new();
+    let mut bcc = Vec::new();
+    let mut subject = Cow::default();
+    let mut body = Cow::default();
+
+    for (key, val) in url.query_pairs() {
+        match key.as_bytes() {
+            b"cc" => {
+                cc.push(val.into());
+            }
+            b"bcc" => {
+                bcc.push(val.into());
+            }
+            b"subject" => {
+                subject = val;
+            }
+            b"body" => {
+                body = val;
+            }
+            _ => (),
+        }
+    }
+
+    let envelope = Envelope {
+        from: vec![ctx.config.address(ctx.account)],
+        to: vec![url.path().to_string()],
+        encoding: ContentTransferEncoding::Base64,
+        bcc: Some(bcc),
+        cc: Some(cc),
+        custom_headers: None,
+        in_reply_to: None,
+        message_id: None,
+        reply_to: None,
+        sender: None,
+        signature: ctx.config.signature(&ctx.account),
+        subject: Some(subject.into()),
```

Comment:
I'd shorten this part by using the `Default` trait of `Envelope`:
```rust
let envelope = Envelope {
      from: vec![ctx.config.address(ctx.account)],
        to: vec![url.path().to_string()],
        encoding: ContentTransferEncoding::Base64,
        bcc: Some(bcc),
        cc: Some(cc),
        signature: ctx.config.signature(&ctx.account),
        subject: Some(subject.into()),
        .. Envelope::default()
   };
```
Agreed?

Reply:
:heavy_check_mark: Done